build-system/tasks/helpers.js

```diff
@@ -457,6 +443,69 @@ function compileJs(srcDir, srcFilename, destDir, options) {
   }
 }
 
+/**
+ * Enables local testing mode for various target files
+ * @param {string} destFilename
+ */
+function maybeEnableLocalTesting(destFilename) {
+  if (process.env.NODE_ENV === 'development') {
+    if (destFilename === 'amp.js') {
+      return enableLocalTesting('dist/amp.js');
+    } else if (destFilename === 'amp-esm.js') {
+      return enableLocalTesting('dist/amp-esm.js');
+    } else if (destFilename === 'amp4ads-v0.js') {
+      return enableLocalTesting('dist/amp4ads-v0.js');
+    } else if (destFilename === 'integration.js') {
+      return enableLocalTesting('dist.3p/current/integration.js');
+    } else if (destFilename === 'amp-shadow.js') {
+      return enableLocalTesting('dist/amp-shadow.js');
+    } else if (destFilename === 'amp-inabox.js') {
+      return enableLocalTesting('dist/amp-inabox.js');
+    } else {
+      return Promise.resolve();
+    }
+  } else {
+    return Promise.resolve();
+  }
+}
+
+/**
+ * Transpiles from TypeScript into intermediary files before compilation and
+ * deletes them afterwards.
+ *
+ * @param {string} srcDir Path to the src directory
+ * @param {string} srcFilename Name of the JS source file
+ * @param {string} destDir Destination folder for output script
+ * @param {?Object} options
+ * @return {!Promise}
+ */
+function compileTs(srcDir, srcFilename, destDir, options) {
+  options = options || {};
+  const startTime = Date.now();
+  transpileTs(srcDir, srcFilename);
+  endBuildStep('Transpiled', srcFilename, startTime);
+  compileJs(srcDir, srcFilename, destDir, options);
+  rimraf.sync(path.join(srcDir, '**/*.js'));
```

Comment:
`transpileTs` and `compileJs` are async and need to be chained.

Reply:
Done in 5e16ff9bcd28d75b583a44240b128546c49e9a50.

Reply:
Bump :)

Reply:
@choumx Okay to merge this?

Reply:
I think `transpileTs` needs to return a Promise.

Reply:
Done in eea99a2e914c140f5f5c4faaa13e5e5815939a4d. PTAL.